spec/apollo-federation/service_field_v2_spec.rb

```diff
@@ -128,6 +128,98 @@ def execute_sdl(schema)
       )
     end
 
+    it 'returns the federation SDL with multiple links for the schema' do
+      product = Class.new(base_object) do
+        graphql_name 'Product'
+
+        field :upc, String, null: false
+      end
+
+      query_obj = Class.new(base_object) do
+        graphql_name 'Query'
+
+        field :product, product, null: true
+      end
+
+      schema = Class.new(base_schema) do
+        query query_obj
+        federation version: '2.3',
+                   links: [{
+                     url: 'https://specs.example.com/federation/v2.3',
+                     import: ['test'],
+                   }]
+      end
+
+      expect(execute_sdl(schema)).to match_sdl(
+        <<~GRAPHQL,
+          extend schema
+            @link(url: "https://specs.apollo.dev/federation/v2.3", import: ["@inaccessible", "@tag", "@composeDirective"])
+            @link(url: "https://specs.example.com/federation/v2.3", import: ["@test"])
+
+          type Product {
+            upc: String!
+          }
+
+          type Query {
+            product: Product
+          }
+        GRAPHQL
+      )
+    end
+
+    it 'returns the federation SDL with compose directives for the schema' do
+      complexity_directive = Class.new(GraphQL::Schema::Directive) do
+        graphql_name 'complexity'
+        argument :fixed, Integer, required: true
+        description 'complexity of the field'
+        locations GraphQL::Schema::Directive::FIELD_DEFINITION
+      end
+
+      product = Class.new(base_object) do
+        graphql_name 'Product'
+
+        field :upc, String, null: false
+      end
+
+      query_obj = Class.new(base_object) do
+        graphql_name 'Query'
+
+        field :product, product, null: true, directives: { complexity_directive => { fixed: 1 } }
```

Comment:
I think this is always going to fail on older GraphQL ruby versions. I think we have a couple options potentially:
1. Drop support for GraphQL Ruby < 1.12 (or maybe even < 2.0)
2. Only run these tests on newer versions
  - If we do this, we should add code to make sure that we warn or raise if composeDirective is used on an older GraphQL Ruby version

I lean towards the first option.

Reply:
@utay Yeah we'd probably want to do it in a separate PR and publish a new major version.

Reply:
@grxy Any plan to do that in the near future? It was the main blocker here I think